src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -150,13 +143,24 @@ public void Prepare(CancellationToken token)
                 }
 
                 RuleSetInformation info = keyValue.Value;
-                RuleSetGroup group = keyValue.Key;
-                Debug.Assert(info.NewRuleSetFilePath == null, "Not expected to be called twice");
+                Debug.Assert(!string.IsNullOrWhiteSpace(info.NewRuleSetFilePath), "Expected to be set during registration time");
 
-                info.NewRuleSetFilePath = this.QueueWriteSolutionLevelRuleSet(this.SolutionFullPath, info.RuleSet, fileNameSuffix: group.ToString());
+                this.sourceControlledFileSystem.QueueFileWrite(info.NewRuleSetFilePath, () =>
+                {
+                    string ruleSetDirectoryPath = Path.GetDirectoryName(info.NewRuleSetFilePath);
+                    if (!this.sourceControlledFileSystem.DirectoryExists(ruleSetDirectoryPath))
+                    {
+                        this.sourceControlledFileSystem.CreateDirectory(ruleSetDirectoryPath);
```

Comment:
No need to check if the directory exists first. Create will be a no-op is it exists.